src/main/com/algos/src/datastructures/caches/LRUCache.java

```diff
@@ -89,18 +79,15 @@ private void moveNodeToLast(Entry<K, V> entry) {
     public void put(K key, V value) {
         if (data.containsKey(key)) {
             final Entry<K, V> existingEntry = data.get(key);
-            existingEntry.setValue(value);
             moveNodeToLast(existingEntry);
+            existingEntry.setValue(value); 
             return;
         }
-        Entry<K, V> newEntry;
-        if (data.size() == cap) {
+        Entry<K, V> newEntry = new Entry<>(); 
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/rishabhbohra3/Algo-tutorial-test/pull/6#issuecomment-2197354925>#c17430</a><span> - 06/28/2024, 05:17 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The initialization of 'newEntry' as a new 'Entry<>' object is redundant because it is immediately reassigned if the cache is at capacity. This can lead to unnecessary object creation, which can impact performance.
</div>
<div id="fix">
<b>Fix:</b> Initialize 'newEntry' only if the cache is not at capacity to avoid unnecessary object creation.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -86,2 +86,2 @@ public void put(K key, V value) {
 -        Entry<K, V> newEntry = new Entry<>();
 +        Entry<K, V> newEntry;
 @@ -90,2 +90,3 @@ public void put(K key, V value) {
 +        } else {
 +            newEntry = new Entry<>();
 -        }
```
</div>
</details>
</div>
<br>


</div>